src/typing_extensions.py

```diff
@@ -1105,17 +1110,7 @@ def __subclasscheck__(cls, other):
 
     _TypedDict = type.__new__(_TypedDictMeta, 'TypedDict', (), {})
 
-    @_ensure_subclassable(lambda bases: (_TypedDict,))
-    def TypedDict(
-        typename,
-        fields=_marker,
-        /,
-        *,
-        total=True,
-        closed=None,
-        extra_items=NoExtraItems,
-        **kwargs
-    ):
+    class TypedDict(metaclass=_TypedDictMeta):
```

Comment:
I suppose it's necessary to make `TypedDict` a class so that you can subscript it? What about making it an instance of a `_SpecialForm` subclass, similar to [how `Annotated` is implemented in CPython](https://github.com/python/cpython/blob/e2476398ee9911b6b0b80e3ca182647805fde81f/Lib/typing.py#L2136-L2195)?

I doubt there's much difference between the two in terms of the user-facing behaviour, but conceptually it sort-of feels "wrong" for `TypedDict` to be a class. You use `TypedDict` to _create_ types; it is not in itself a type; calling `TypedDict` does not create an "instance of `TypedDict`"; `x: TypedDict` is an invalid type annotation; etc.

Reply:
I'll give it a try.

Reply:
I opened https://github.com/python/typing_extensions/pull/580 to compare both.